src/librustdoc/lib.rs

```diff
@@ -814,7 +814,12 @@ fn main_args(early_dcx: &mut EarlyDiagCtxt, at_args: &[String]) {
         }
     };
 
-    match (options.should_test, config::markdown_input(&input)) {
+    let output_format = options.output_format;
+
+    match (
+        options.should_test || output_format == config::OutputFormat::Doctest,
+        config::markdown_input(&input),
+    ) {
         (true, Some(_)) => return wrap_return(dcx, doctest::test_markdown(&input, options)),
         (true, None) => return doctest::run(dcx, input, options),
```

Comment:
`doctest::run` should be renamed, because now it either runs *or* just collects and prints the doctests.

Reply:
We run the `doctest` pass. Same for outputs named `run_renderer`. I don't mind changing it though, how would you rename it?